hpx/parallel/algorithms/count.hpp

```diff
@@ -48,9 +107,21 @@ namespace hpx { namespace parallel { HPX_INLINE_NAMESPACE(v1)
 
             template <typename ExPolicy, typename Iter, typename T>
             static difference_type
-            sequential(ExPolicy, Iter first, Iter last, T const& value)
+            sequential(ExPolicy && policy, Iter first, Iter last, T const& value)
             {
-                return std::count(first, last, value);
+                auto f1 =
+                    count_iteration<ExPolicy, detail::compare_to<T> >(
+                        std::forward<ExPolicy>(policy),
+                        detail::compare_to<T>(value));
+
+                using hpx::util::placeholders::_1;
+                typename std::iterator_traits<Iter>::difference_type ret = 0;
+
+                util::loop(
+                    policy, first, last,
+                    hpx::util::bind(std::move(f1), _1, std::ref(ret)));
```

Comment:
Only if we define yet another type... Or do I misunderstand what you have in mind?